File: app/src/main/java/ru/SnowVolf/convertx/fragments/extended/UnescaperFragment.java
package ru.SnowVolf.convertx.fragments.extended;

import android.graphics.Typeface;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.view.menu.MenuBuilder;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import ru.SnowVolf.convertx.R;
import ru.SnowVolf.convertx.algorhitms.DecoderAlgs;
import ru.SnowVolf.convertx.other.Extras;
import ru.SnowVolf.convertx.ui.Toasty;

import static ru.SnowVolf.convertx.utils.StringUtils.copyToClipboard;

/**
 * Created by Snow Volf on 22.02.2017, 19:51
 */

public class UnescaperFragment extends Fragment {
    public EditText data;
    public TextView dataOut;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState){
        View rootView = inflater.inflate(R.layout.fragment_ex, container, false);
        data = (EditText) rootView.findViewById(R.id.exData);
        data.setHint("Текст зашифрованого XML");
        data.setFocusable(true);
        dataOut = (TextView) rootView.findViewById(R.id.exText);
        dataOut.setHint("Преобразует символы \"&quot;\", \"&apos;\", т.д. в нормальный текст");
        return rootView;
    }
    @Override
    public void onDestroyView(){
        super.onDestroyView();
    }
    @Override
    public void onCreate(Bundle savedInstanceState){
        super.onCreate(savedInstanceState);
        //setRetainInstance(true);
        setHasOptionsMenu(true);
    }

    public void onActivityCreated(Bundle savedInstanceState){
        super.onActivityCreated(savedInstanceState);
        Typeface Mono = Typeface.createFromAsset(getActivity().getAssets(), "fonts/RobotoMono-Regular.ttf");
        data.setTypeface(Mono);
        dataOut.setTypeface(Mono);
        data.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
            }

            @Override
            public void afterTextChanged(Editable editable) {
                String source = data.getText().toString();
                String unescaped = DecoderAlgs.unescapeXml(source);
                dataOut.setText(unescaped);
            }
        });
    }

    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        if (menu != null) {
            menu.clear();
        } else
            menu = new MenuBuilder(getContext());
        //добавляем пункты меню
        menu.add(R.string.copy2clipboard)
                .setIcon(R.drawable.ic_menu_copy)
                .setShowAsActionFlags(MenuItem.SHOW_AS_ACTION_ALWAYS)
                .setOnMenuItemClickListener(menuItem -> {
                    copyToClipboard(getContext(), dataOut.getText().toString());
                    Extras.showToast(getContext(), getString(R.string.copied2clipboard));
                    return true;
                });
        menu.add(R.string.clear_all)
                .setIcon(R.drawable.ic_menu_clear_all)
                .setShowAsActionFlags(MenuItem.SHOW_AS_ACTION_ALWAYS)
                .setOnMenuItemClickListener(menuItem -> {
                    clearAllText();
                    return true;
                });
    }
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home)
            getActivity().finish();
        return true;
    }

    public void clearAllText(){
        data.setText("");
        dataOut.setText("");
        Toasty.success(getContext(), getString(R.string.cleared), Toast.LENGTH_SHORT, true).show();
    }

}

File: app/src/main/java/ru/SnowVolf/convertx/ui/fragments/extended/UnescaperFragment.kt
package ru.SnowVolf.convertx.ui.fragments.extended

import android.content.Context
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.*
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.view.menu.MenuBuilder
import ru.SnowVolf.convertx.R
import ru.SnowVolf.convertx.algorhitms.DecoderAlgs
import ru.SnowVolf.convertx.settings.Preferences
import ru.SnowVolf.convertx.ui.Toasty
import ru.SnowVolf.convertx.ui.fragments.base.BaseFragment
import ru.SnowVolf.convertx.utils.StringUtils

/**
 * Created by Snow Volf on 22.02.2017, 19:51
 */

class UnescaperFragment : BaseFragment() {
    lateinit var data: EditText
    lateinit var dataOut: TextView

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        rootView = inflater?.inflate(R.layout.fragment_ex, container, false)
        data = rootView?.findViewById(R.id.exData) as EditText
        data.hint = "Текст зашифрованого XML"
        data.isFocusable = true
        dataOut = rootView?.findViewById(R.id.exText) as TextView
        dataOut.hint = "Преобразует символы \"&quot;\", \"&apos;\", т.д. в нормальный текст"
        return rootView
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setRetainInstance(true);
        setHasOptionsMenu(true)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        TITLE = R.string.xml
        data.typeface = Mono
        data.textSize = Preferences.getFontSize().toFloat()
        dataOut.typeface = Mono
        data.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}

            override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}

            override fun afterTextChanged(editable: Editable) {
                val source = data.text.toString()
                val unescaped = DecoderAlgs.unescapeXml(source)
                dataOut.text = unescaped
            }
        })
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        var menu = menu
        if (menu != null) {
            menu.clear()
        } else
            menu = MenuBuilder(context)
        //добавляем пункты меню
        menu.add(R.string.copy2clipboard)
            .setIcon(R.drawable.ic_menu_copy)
            .setShowAsActionFlags(MenuItem.SHOW_AS_ACTION_ALWAYS)
            .setOnMenuItemClickListener { menuItem ->
                StringUtils.copyToClipboard(requireContext(), dataOut.text.toString())
                showToast(context, getString(R.string.copied2clipboard))

                true
            }
        menu.add(R.string.clear_all)
            .setIcon(R.drawable.ic_menu_clear_all)
            .setShowAsActionFlags(MenuItem.SHOW_AS_ACTION_ALWAYS)
            .setOnMenuItemClickListener { menuItem ->
                clearAllText()
                true
            }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item!!.itemId == android.R.id.home)
            requireActivity().finish()
        return true
    }

    fun clearAllText() {
        data.setText("")
        dataOut.text = ""
        Toasty.success(requireContext(), getString(R.string.cleared), Toast.LENGTH_SHORT, true).show()
    }

    fun showToast(context: Context?, text: String?) {
        Toast.makeText(context, text, Toast.LENGTH_SHORT).show()
    }

}
